src/commands/bench.rs

```diff
@@ -183,18 +184,22 @@ fn bench_revset<M: Measurement>(
 ) -> Result<(), CommandError> {
     writeln!(ui, "----------Testing revset: {revset}----------\n")?;
     let expression = workspace_command.parse_revset(revset)?;
-    // Time both evaluation and iteration. Note that we don't clear caches (such as
-    // commit objects in `Store`) between each run (`criterion`
-    // doesn't seem to support that).
-    let routine = |expression| {
-        workspace_command
-            .evaluate_revset(expression)
-            .unwrap()
-            .iter()
-            .count()
+    // Reload repo to clear caches (such as commit objects in `Store`), but preload
+    // index since it's more likely to be loaded by preceding operation.
+    let reload_repo = || {
+        let repo = workspace_command.repo();
+        let repo = repo.reload_at(repo.operation());
```

Comment:
Despite the name, this reuses the different backends. I think the easiest way to reload the repo and stores is to create a new WorkspaceCommandHelper from the CommandHelper

Reply:
Oops, that's why `author(_)` was fast than I expected. I'll rewrite it to use `command.workspace_helper_no_snapshot()`.